host/lib/utils/paths.cpp

```diff
@@ -173,10 +177,28 @@ std::string uhd::get_app_path(void){
     return uhd::get_tmp_path();
 }
 
+#if BOOST_VERSION >= 106100
+std::string uhd::get_lib_path(void) {
+    fs::path runtime_libfile_path = boost::dll::this_line_location();
+    return runtime_libfile_path.remove_filename().string();
+}
+
+std::string uhd::get_pkg_path(void) {
+    fs::path pkg_path = fs::path(uhd::get_lib_path()).parent_path().lexically_normal();
+    return get_env_var("UHD_PKG_PATH", pkg_path.string());
+}
+#else
 std::string uhd::get_pkg_path(void) {
     return get_env_var("UHD_PKG_PATH", UHD_PKG_PATH);
 }
 
+std::string uhd::get_lib_path(void) {
+    fs::path lib_path = fs::path(uhd::get_pkg_path()) / UHD_LIB_DIR;
+    return lib_path.string();
+}
+#endif
```

Comment:
I've opted for different versions of both ``get_lib_path`` and ``get_pkg_path`` depending on the boost version because it felt cleaner. The dynamic method currently allow the user to override the result of ``get_pkg_path`` but not ``get_lib_path`` with the ``UHD_PKG_PATH`` environment variable, but the hard-coded method lets the environment variable affect both. I don't know if that's desired behavior for the dynamic method, but the difference makes it cleaner to just redefine both functions.